### PR TITLE
Update to esp-hal@0.22 and esp-wifi@0.9

### PR DESCRIPTION
**Pull request by JurajSadel**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -6,25 +6,48 @@ edition = "2021"
...
+] }
+esp-wifi = { version = "0.11.0", default-features = false, features = [
+    "esp32c3",
+    "utils",
+    "wifi",
+    "esp-alloc",
+    "log",
+] }
+esp-alloc = { version = "0.5.0" }
+log = "0.4.22"
```
Comment (SergioGasquez) [inline Cargo.toml:20]: Are you using log? Briefly looking, it's not used, it might be a good idea to switch from `println` to logs.

**Review by SergioGasquez — APPROVED** [review]
LGTM! Only one small question

File: Cargo.toml [hunk 2]
```diff
@@ -6,25 +6,48 @@ edition = "2021"
...
+embassy-executor = { version = "0.6.3", features = [
     "nightly",
     "integrated-timers",
 ] }
-embassy-time       = { version = "0.3.0" }
-embedded-hal       = { version = "0.2.7", features = ["unproven"] }
-embedded-hal-1     = { version = "1.0.0", package = "embedded-hal" }
+embassy-time = { version = "0.3.2" }
+embedded-hal = { version = "0.2.7", features = ["unproven"] }
+embedded-hal-1 = { version = "1.0.0", package = "embedded-hal" }
```
Comment (SergioGasquez) [inline Cargo.toml]: I think we should also make eh1 the default (that is naming it `embedded-hal` and using `embedded-hal-02` for eh02)

Comment (JurajSadel) [reply Cargo.toml]: Thanks!

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks for updating it!

**Pull request merged**